src/Core/Services/VaultTimeoutService.cs

```diff
@@ -146,7 +146,7 @@ public async Task LockAsync(bool allowSoftLock = false, bool userInitiated = fal
 
             if (await _keyConnectorService.GetUsesKeyConnector()) {
                 var pinSet = await IsPinLockSetAsync(userId);
-                var pinLock = (pinSet.Item1 && _stateService.GetPinProtectedAsync(userId) != null) || pinSet.Item2;
+                var pinLock = (pinSet.Item1 && await _stateService.GetPinProtectedKeyAsync(userId) != null) || pinSet.Item2;
```

Comment:
Could we use explicit names here instead of `Item1` and `Item2`?
instead of 
`pinSet = await IsPinLockSetAsync();` 
you could have
`(isPinProtected, isPinProtectedWithKey) = await IsPinLockSetAsync();`